site/src/components/mdx/image/Image.tsx

```diff
@@ -9,14 +10,27 @@ export interface ImageProps extends Omit<NextImageProps, "src"> {
    * Image source url
    */
   src: string;
+  /**
+   * Image source url for dark mode.
+   */
+  srcDark?: string;
 }
 
 export const Image = forwardRef<HTMLDivElement, ImageProps>(function Image(
-  { className, src, fill, width, height, alt, unoptimized, ...rest },
+  { className, src, srcDark, fill, width, height, alt, unoptimized, ...rest },
   ref,
 ) {
-  const resolvedSrc = useResolveRelativeUrl(src);
+  const siteMode = useColorMode();
+
+  const srcToUse = siteMode === "dark" && srcDark ? srcDark : src;
+
+  const resolvedSrc = useResolveRelativeUrl(srcToUse);
+
+  const resolvedSrcToUse =
```

Comment:
FYI: Did look at this
https://nextjs.org/docs/pages/api-reference/components/image#theme-detection-css
but really only of benefit when using 
`@media (prefers-color-scheme: dark) {`
Whilst we have an imperative  theme switcher, still good to know that NextImage has additional props/patterns of use.

Reply:
Approach mentioned will load 2 images regardless of modes. The best native HTML approach is by using [`<picture>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture), however media query in `<source>` doesn't work with custom theme toggle